reviews three/swagger.yaml

```diff
@@ -0,0 +1,954 @@
+      discount:
+          type: object
+          $ref: '#/definitions/discount'
+      tax_line:
+          type: array
+          items:
+            $ref: '#/definitions/tax_line'
+      
+        
+  custom_line_item:
+    description: custom line item
+    allOf:
+    - $ref: '#/definitions/line_item'
+    - type: object
+      required:
+      - title
+      - price
+      - grams
+    
+      properties:
+        title:
+          type: string
+          description:  The title of the product or variant.  Can only be specified if this is a custom line item, in which case it is mandatory.
+        price:
+          type: string
+          description: The price of the item before discounts have been applied.  Can only be specified if this is a custom line item, in which case it is mandatory.  To change the price of a product, use `discount` as specified below.
+        grams:
+          type: string
+          description: Weight in grams.  Can only be specified if this is a custom line item.  If not specified, defaults to `0`.
+        requires_shipping:
+          type: boolean
+          description: States whether or not the fulfillment requires shipping. Values are `true` or `false`.  Can only be specified if this is a custom line item.  Defaults to `false`.
+          enum: 
+          - true
+          - false
+        sku:
+          type: string
+          description: Can only be specified if this is a custom line item.
+        taxable:
+          type: boolean
+          description: States whether or not the product is taxable. Values are `true` or `false`.  Can only be specified if this is a custom line item.  Mandatory for custom products.
+
+  product_variant_line_item:      
+    description: product variant line item
+    allOf:
+    - $ref: '#/definitions/line_item'
+    - type: object
+      required:
+      - variant_id
+       
+      properties:
+        variant_id:
+          type: string
+          description: The `id` of the product variant corresponding to the `line_item`.  Null if custom `line_item`.
+          readOnly: false
+        product_id:
+          type: string
+          description: The `id` of the product corresponding to the line item’s product variant.
+        name:
+          type: string
+          description: The readable name of the product and variant.
+        variant_title:
+          type: string
+          description: The title of the product variant.  Default value is “Custom”.
+
+    
+  
+  shipping_rate:
+    type: object
+    required:
+    - handle
+    - title
+    - price
+    properties:
+      handle:
+        type: string
+        description: handle of the shipping rate.
+      title:
+        type: string
+        description: The shipping method name.
+      price:
+        type: string
+        description: The price of the shipping rate.
+
+
+  shipping_line:
+    type: object
+    required:
+    - title
+    - price
+    - tax_lines
```

Comment:
tax_line sis read only and not required. A read only field cannot be mandatory, its completely impossible. Can you cross reference?